File: gams/apifiles/C++/Transport1/Transport1.cpp
// Transport1.cpp : main project file.

#include "stdafx.h"

using namespace System;
using namespace System::IO;
using namespace GAMS;

int main(array<String ^> ^args)
{
	GAMSWorkspace^ ws;
	if (args->Length > 0) 
		ws = gcnew GAMSWorkspace(nullptr,args[0],false);
	else
		ws = gcnew GAMSWorkspace(nullptr,nullptr,false);
	
	ws->GamsLib("trnsport");
	GAMSJob^ t1 = ws->AddJobFromFile("trnsport.gms",nullptr,nullptr);

	t1->Run();
	Console::WriteLine(L"Ran with Default:");

	for each (GAMSVariableRecord^ rec in t1->OutDB->GetVariable("x"))
		Console::WriteLine("x(" + rec->Keys[0] + "," + rec->Keys[1] + "): level=" + rec->Level + " marginal=" + rec->Marginal);

	// run the job again with another solver
	{
		GAMSOptions^ opt = ws->AddOptions(nullptr);
		{
			opt->AllModelTypes = "xpress";
			t1->Run(opt);
		}
	}

	Console::WriteLine("Ran with XPRESS:");
	for each (GAMSVariableRecord^ rec in t1->OutDB->GetVariable("x"))
		Console::WriteLine("x(" + rec->Keys[0] + "," + rec->Keys[1] + "): level=" + rec->Level + " marginal=" + rec->Marginal);

	// run the job with a solver option file
	{
		StreamWriter^ optFile = gcnew StreamWriter(Path::Combine(ws->WorkingDirectory, "xpress.opt"));
		GAMSOptions^ opt = ws->AddOptions(nullptr);
		{
			optFile->WriteLine("algorithm=barrier");
			optFile->Close();
			opt->AllModelTypes = "xpress";
			opt->OptFile = 1;
			t1->Run(opt);
		}
	}

	Console::WriteLine("Ran with XPRESS with non-default option:");
	for each (GAMSVariableRecord^ rec in t1->OutDB->GetVariable("x"))
		Console::WriteLine("x(" + rec->Keys[0] + "," + rec->Keys[1] + "): level=" + rec->Level + " marginal=" + rec->Marginal);
	return 0;
}

File: gams/apifiles/CSharp/Transport1/Transport1.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Text;
using GAMS;

namespace TransportSeq
{
    class Transport1
    {
        static void Main(string[] args)
        {
            GAMSWorkspace ws;
            if (Environment.GetCommandLineArgs().Length > 1)
                ws = new GAMSWorkspace(systemDirectory: Environment.GetCommandLineArgs()[1]);
            else
                ws = new GAMSWorkspace();
            ws.GamsLib("trnsport");

            // create a GAMSJob from file and run it with default settings
            GAMSJob t1 = ws.AddJobFromFile("trnsport.gms");
            
            t1.Run();
            Console.WriteLine("Ran with Default:");
            foreach (GAMSVariableRecord rec in t1.OutDB.GetVariable("x"))
                Console.WriteLine("x(" + rec.Keys[0] + "," + rec.Keys[1] + "): level=" + rec.Level + " marginal=" + rec.Marginal);

            // run the job again with another solver
            using (GAMSOptions opt = ws.AddOptions())
            {
                opt.AllModelTypes = "xpress";
                t1.Run(opt);
            }
            Console.WriteLine("Ran with XPRESS:");
            foreach (GAMSVariableRecord rec in t1.OutDB.GetVariable("x"))
                Console.WriteLine("x(" + rec.Keys[0] + "," + rec.Keys[1] + "): level=" + rec.Level + " marginal=" + rec.Marginal);

            // run the job with a solver option file
            using (StreamWriter optFile = new StreamWriter(Path.Combine(ws.WorkingDirectory, "xpress.opt")))
            using (GAMSOptions opt = ws.AddOptions())
            {
                optFile.WriteLine("algorithm=barrier");
                optFile.Close();
                opt.AllModelTypes = "xpress";
                opt.OptFile = 1;
                t1.Run(opt);
            }
            Console.WriteLine("Ran with XPRESS with non-default option:");
            foreach (GAMSVariableRecord rec in t1.OutDB.GetVariable("x"))
                Console.WriteLine("x(" + rec.Keys[0] + "," + rec.Keys[1] + "): level=" + rec.Level + " marginal=" + rec.Marginal);
        }
    }
}
